src/components/molecules/NavList.tsx

```diff
@@ -1,53 +1,187 @@
-import React from 'react';
-import {List, ListItemButton, ListItemText, ListSubheader} from '@mui/material';
+import React, { useState } from "react";
+import {
+  List,
+  ListItemButton,
+  ListItemText,
+  ListSubheader,
+} from "@mui/material";
+import { Link } from "react-router-dom";
+
+const commonStyle = {
+  fontWeight: "bold",
+  bgcolor: "black",
+  color: "white",
+  fontSize: 20,
+};
+
+function generateButtonStyle(isSelected: boolean) {
+  return {
+    ...(isSelected && {
+      backgroundColor: "lightgrey",
+      padding: "1px 16px",
+      borderRadius: "0 24px 24px 0",
+      "&:hover": {
+        backgroundColor: "lightgrey",
+      },
+    }),
+  };
+}
+
+function NavList() {
+  const [selectedLink, setSelectedLink] = useState<string | null>(null);
+
+  const handleLinkClick = (link: string) => {
+    setSelectedLink(link);
+  };
+
+  const isLinkSelected = (link: string) => selectedLink === link;
```

Comment:
Can the color that this typically sets be included in the `generateButtonStyle` function?

Reply:
No idea, but possibly!

Reply:
The answer is yes!